Source/Fuse.Storage/AndroidUserSettings.uno

```diff
@@ -0,0 +1,124 @@
+using Uno;
+using Uno.UX;
+using Uno.Compiler.ExportTargetInterop;
+
+namespace Fuse.Storage
+{
+	[ForeignInclude(Language.Java, "android.content.SharedPreferences", "android.preference.PreferenceManager")]
+	internal extern(Android) class AndroidUserSettings : IUserSettings
+	{
+		Java.Object _handle;
+		Java.Object _handleEditor;
+
+		public AndroidUserSettings()
+		{
+			_handle = Init();
+			_handleEditor = GetEditor(_handle);
+		}
+
+		[Foreign(Language.Java)]
+		Java.Object Init()
+		@{
+			return PreferenceManager.getDefaultSharedPreferences(com.fuse.Activity.getRootActivity());
+		@}
+
+		[Foreign(Language.Java)]
+		Java.Object GetEditor(Java.Object handle)
+		@{
+			return ((SharedPreferences)handle).edit();
+		@}
+
+		public string GetStringValue(string key)
+		{
+			return GetStringValue(_handle, key);
+		}
+
+		[Foreign(Language.Java)]
+		string GetStringValue(Java.Object handle, string key)
+		@{
+			return ((SharedPreferences)handle).getString(key, "");
+		@}
```

Comment:
OK, so here you have a separate method just so you can easily pass _handle. This gets a bit more tedious to read than it has to, IMO. A pattern I've used in the past is to have an internal static class, called something like `SharedPreferencesHelpers`, and putting all the foreign code on that. This reduces context-switching a bit when reading the code. You can see [examples of this](https://github.com/fuse-open/uno/blob/master/lib/Uno.Threading/Implementation/Posix/PthreadHelpers.uno) in our threading code.

Another alternative is to drop the Uno-wrappers, and use `@{AndroidUserSettings:Of(_this)._handle:Get()}` to get _handle instead of passing it in.

A final option is to simply just move all the helpers to the top of the source-file and call it the day :)

This is all largely taste-based, so I leave it up to you, just wanted to make sure you're aware of these options.

Reply:
Ok I'll try to simplifly it by removing unnecessary uno wrapper. thanks